src/main/java/org/jabref/model/entry/identifier/SSRN.java

```diff
@@ -0,0 +1,78 @@
+package org.jabref.model.entry.identifier;
+
+import java.net.URI;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Represents an SSRN article, identified by its abstract ID.
+ */
+public class SSRN extends EprintIdentifier {
+
+    private static final String SSRN_URL_REGEX = "(https?://)?(papers\\.)?ssrn\\.com/(sol3/papers.cfm\\?)?abstract(_id)?=(?<id>\\d+)";
+    private static final Pattern SSRN_URL_FULL_MATCH = Pattern.compile("^" + SSRN_URL_REGEX + "$", Pattern.CASE_INSENSITIVE);
+    private static final Pattern SSRN_URL_MATCH = Pattern.compile(SSRN_URL_REGEX, Pattern.MULTILINE | Pattern.CASE_INSENSITIVE);
+
+    private final int abstractId;
+
+    /**
+     * Tries to parse an SSRN identifier
+     *
+     * @param string Either a number or a SSRN url that has the abstract ID in it
+     * @throws NullPointerException If you pass a null to the constructor
+     * @throws IllegalArgumentException Invalid string passed to the constructor
+     */
+    public SSRN(String string) {
+        Objects.requireNonNull(string);
+        string = string.trim();
+
+        if (string.chars().allMatch(Character::isDigit)) {
+            abstractId = Integer.parseInt(string);
+            return;
+        }
+
+        Matcher matcher = SSRN_URL_FULL_MATCH.matcher(string);
+        if (matcher.find()) {
+            abstractId = Integer.parseInt(matcher.group("id"));
+            return;
+        }
+
+        throw new IllegalArgumentException(string + " is not a valid SSRN identifier");
+    }
+
+    public SSRN(int abstractId) {
+        this.abstractId = abstractId;
+    }
+
+    public static Optional<SSRN> parse(String data) {
+        Matcher matcher = SSRN_URL_MATCH.matcher(data);
+        if (matcher.find()) {
+            int abstractId = Integer.parseInt(matcher.group("id"));
+            return Optional.of(new SSRN(abstractId));
+        }
+
+        return Optional.empty();
+    }
+
+    @Override
+    public String getNormalized() {
+        return String.valueOf(abstractId);
+    }
+
+    @Override
+    public Optional<URI> getExternalURI() {
+        URI uri = URI.create("https://ssrn.com/abstract=" + abstractId);
+        return Optional.of(uri);
+    }
+
+    /**
+     * Generate the DOI of the article
+     *
+     * @return a short doi formatted as "10.2139/ssrn.[article]"
```

Comment:
```suggestion
     * @return a DOI formatted as <code>10.2139/ssrn.[article]</code>
```

short DOI is something different: https://shortdoi.org/ -- OMG, we have it wrongly in `DOI.java`. Needs to be fixed in a follow up. Maybe use "plain DOI" and "DOI with HTTP URL"

Reply:
I won't update the terminology elsewhere, but just "DOI" for the identifier and "DOI proxy" for the URL seems to be the terminology they use on their website.